app/src/test/java/org/oppia/android/app/testing/PlatformParameterIntegrationTest.kt

```diff
@@ -102,29 +147,168 @@ class PlatformParameterIntegrationTest {
+      setUpApplicationForContext(MockPlatformParameterService.appVersionForCorrectResponse)
+      platformParameterController.updatePlatformParameterDatabase(
+        mockPlatformParameterListWithToastDisabled
+      )
+
+      val workManager = WorkManager.getInstance(context)
+      val requestId = setUpAndEnqueueSyncUpWorkerRequest(workManager)
+      testCoroutineDispatchers.runCurrent()
+
+      val workInfo = workManager.getWorkInfoById(requestId)
+      // Check the work request succeeded which means the local database was updated with new values.
+      assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      scenario.onActivity { activity ->
+        activity.splashTestActivityPresenter.loadPlatformParameters()
+      }
+      testCoroutineDispatchers.runCurrent()
+
+      // As the local database was updated correctly the app will use the server values, and the
+      // server value for the splash screen welcome msg param is true.
       assertThat(ShadowToast.getLatestToast()).isNotNull()
       assertThat(ShadowToast.getTextOfLatestToast()).isEqualTo(SplashTestActivity.WELCOME_MSG)
     }
   }
 
+  @Test
+  fun testIntegration_executeSyncUpWorkerIncorrectly_readDatabase_checkWelcomeMsgIsInvisible() {
+    launch(SplashTestActivity::class.java).use { scenario ->
+      // Set up versionName to get incorrect network response from mock platform parameter service.
+      setUpApplicationForContext(MockPlatformParameterService.appVersionForWrongResponse)
+      platformParameterController.updatePlatformParameterDatabase(
+        mockPlatformParameterListWithToastDisabled
+      )
+
+      val workManager = WorkManager.getInstance(context)
+      val requestId = setUpAndEnqueueSyncUpWorkerRequest(workManager)
+      testCoroutineDispatchers.runCurrent()
+
+      val workInfo = workManager.getWorkInfoById(requestId)
+      // Check the work request fails because of incorrect network response. This means that the
+      // local database is not updated with new values.
+      assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.FAILED)
+
+      // Fetch the latest platform parameter from cache store after execution of work request to
+      // imitate the loading process at the start of splash test activity.
+      scenario.onActivity { activity ->
+        activity.splashTestActivityPresenter.loadPlatformParameters()
+      }
+      testCoroutineDispatchers.runCurrent()
+
+      // As the local database was not updated due to work request failure the app will use default
+      // values, and the default value for the splash screen welcome msg param is false.
+      assertThat(ShadowToast.getLatestToast()).isNull()
+    }
+  }
+
+  private fun setUpApplicationForContext(testAppVersionName: String) {
```

Comment:
shouldn't be the name says `setUpApplicationForVersionName` ?

Reply:
Yeah. Done